pkg/controllers/job/job_controller.go

```diff
@@ -215,20 +241,69 @@ func (cc *Controller) Run(stopCh <-chan struct{}) {
 	glog.Infof("JobController is running ...... ")
 }
 
-func (cc *Controller) worker() {
-	for cc.processNextReq() {
+func (cc *Controller) worker(i uint32) {
+
+	glog.Infof("worker %d start ...... ", i)
+
+	for cc.processNextReq(i) {
+	}
+}
+
+// TODO we may need to make this sharding more proper if required
+func (cc *Controller) belongsToThisRoutine(key string, count uint32) bool {
+	var hashVal hash.Hash32
+	var val uint32
+
+	hashVal = fnv.New32()
+	hashVal.Write([]byte(key))
+
+	val = hashVal.Sum32()
+
+	if val%cc.workers == count {
+		return true
 	}
+
+	return false
+}
+
+// TODO we may need to make this sharding more proper if required
+func (cc *Controller) getWorkerID(key string) workqueue.RateLimitingInterface {
```

Comment:
This function doesn't work as its name describes

Reply:
changed the name